pye57/e57.py

```diff
@@ -170,20 +194,29 @@ def read_scan(self,
         data, buffers = self.make_buffers(fields, n_points)
         header.points.reader(buffers).read()
 
-        valid = ~data["cartesianInvalidState"].astype("?")
+        if validState in data:
+            valid = ~data[validState].astype("?")
 
-        for field in data:
-            data[field] = data[field][valid]
+            for field in data:
+                data[field] = data[field][valid]
 
-        del data["cartesianInvalidState"]
+            del data[validState]
 
         if transform:
-            xyz = np.array([data["cartesianX"], data["cartesianY"], data["cartesianZ"]]).T
-            xyz = self.to_global(xyz, header.rotation, header.translation)
+            if coordinate_system is COORDINATE_SYSTEMS.CARTESIAN:
+                xyz = np.array([data["cartesianX"], data["cartesianY"], data["cartesianZ"]]).T
+                if header.has_pose():
+                    xyz = self.to_global(xyz, header.rotation, header.translation)
+            elif coordinate_system is COORDINATE_SYSTEMS.SPHERICAL:
+                rae = np.array([data["sphericalRange"], data["sphericalAzimuth"], data["sphericalElevation"]]).T
+                # TODO: global tranform for spherical cooridinates
```

Comment:
Does this comment refer to the case when a scan has a pose, and is in spherical coordinates? From what I undestand, I think this case is not handled here...

Reply:
yes, that was the case. But updated it now.

1. `xyz` in calculated in both coordinate system
2. Transformed to global if pose is available